python/ray/serialization.py

```diff
@@ -218,12 +233,8 @@ def _deserialize_object(self, data, metadata, object_ref):
             elif error_type == ErrorType.Value("WORKER_DIED"):
                 return WorkerCrashedError()
             elif error_type == ErrorType.Value("ACTOR_DIED"):
-                if data:
-                    pb_bytes = self._deserialize_msgpack_data(
-                        data, metadata_fields)
-                    if pb_bytes:
-                        return RayError.from_bytes(pb_bytes)
-                return RayActorError()
+                return self._deserialize_actor_died_error(
+                    data, metadata_fields)
```

Comment:
Can we remove error_type? Since we have a RayError PB schema here,  we can move error_type into RayError, instead of appending an meta field to the bytes. That will make code cleaner and more matainable.

Reply:
You mean `error_type == ErrorType.Value("ACTOR_DIED"):` -> this one?

It is too big scope to handle here. Consider creating an issue?

Reply:
But agreed this could be cleaned up by being included within the protobuf